src/core/qgsmaplayer.h

```diff
@@ -1102,10 +1102,11 @@ class CORE_EXPORT QgsMapLayer : public QObject
      * record in the users style table in their personal qgis.db)
      * \param resultFlag a reference to a flag that will be set to FALSE if
      * we did not manage to save the default style.
+     * \param categories the style categories to be saved.
      * \returns a QString with any status messages
      * \see loadNamedStyle() and \see saveNamedStyle()
      */
-    virtual QString saveDefaultStyle( bool &resultFlag SIP_OUT );
+    virtual QString saveDefaultStyle( bool &resultFlag SIP_OUT, StyleCategories categories = AllStyleCategories );
```

Comment:
This is technically an API break -- eg a QgsPluginLayer subclass implementing an override will no longer work correctly. Are you able to deprecate this method instead and add a new virtual method so that API is maintained?

Reply:
I chose to deprecate this one-parameter method, so in the future it will be a two-parameters method (with the second parameter having a default value).

But for now, to avoid an API break, I see no other solution than to stick to 2 signatures : 
- a one-parameter method, deprecated
- a two-parameters method

When the one-parameter method will be deleted, the second parameter of the two-parameter method will have to have a default value. As a result, the callers will still be able to call this method with one parameter only like most of them today.

So I had to adjust the code to prevent the calls to the one-parameter method to create deprecated warnings today, because in the future it will still be the right way to call the method. (see https://github.com/qgis/QGIS/pull/48394/files#diff-6d3f71e37dccdeaa6ee39db730250e9d9d9105cb168ec9d8379a1df3b4a4569eR235 for example)

I am obviously open to your suggestions on this...